src/library.js

```diff
@@ -1205,14 +1206,17 @@ LibraryManager.library = {
   // an invalid index into the FUNCTION_TABLE, so something has gone wrong.
   __cxa_end_catch__deps: ['__cxa_free_exception', '$EXCEPTIONS'],
   __cxa_end_catch: function() {
-    if (___cxa_end_catch.rethrown) {
-      ___cxa_end_catch.rethrown = false;
+    var ptr = EXCEPTIONS.caught.pop();
+    if (ptr && EXCEPTIONS.infos[EXCEPTIONS.deAdjust(ptr)].rethrown) {
```

Comment:
also, by not calling decref here, the refcount can be zero but the exception isn't cleaned up - is that intentional?